ssc/common.cpp

```diff
@@ -1077,25 +1078,25 @@ bool shading_factor_calculator::use_shade_db()
 	return (m_enTimestep && (m_string_option == 0)); // determine which fbeam function to call
 }
 
-size_t shading_factor_calculator::get_row_index_for_input(size_t hour, size_t hour_step, size_t steps_per_hour)
+size_t shading_factor_calculator::get_row_index_for_input(size_t hour_of_year, size_t minute)
 {
-	// handle different simulation timesteps and shading input timesteps
-	size_t ndx = hour * m_steps_per_hour; // m_beam row index for input hour
-	int hr_step = 0;
-	if (steps_per_hour > 0)
-		hr_step = (int)((int)m_steps_per_hour*(int)hour_step/ (int)steps_per_hour);
-	if (hr_step >= m_steps_per_hour) hr_step = m_steps_per_hour - 1;
-	if (hr_step < 0) hr_step = 0;
-	ndx += hr_step;
+	// this function needs to handle different simulation timesteps and timeseries shading input timesteps
+	// therefore, start the row index by multiplying the hour of the year by the number of timesteps per hour in the timeseries shading input
+	size_t ndx = hour_of_year * (size_t)m_steps_per_hour;
+
+	// then figure out how many row indices to add for the subhourly timeseries entries based on the minute stamp
+	ndx += floor((int)minute / (60 / (m_steps_per_hour)));
+
```

Comment:
How does this work if the minute column is specified as 30 for an hourly weather file? 
The CMPvsamv1PowerIntegration_cmod_pvsamv1.NoFinancialModelShading tests were failing with this change previously.
![image](https://user-images.githubusercontent.com/6498311/86761448-32ec4000-c003-11ea-8d6b-c26f635e1145.png)

Reply:
This is a different implementation of this logic than was used previously, and all tests pass with this logic (including CMPvsamv1PowerIntegration_cmod_pvsamv1.NoFinancialModelShading). Added a comment to explain the logic better, using the 30 minute example.